File: Day1/first.py
numbers = []
res = 0

with open('input1.txt') as file:
    for line in file:
        aux = []
        for letter in line:
            if letter.isdigit():
                aux.append(int(letter))
        numbers.append(str(aux[0]) + str(aux[len(aux)-1]))

for num in numbers:
    res += int(num)

print(f"The sum of all values is: {res}")

File: Day1/first.rb
begin
    file = File.open("example1.txt", "r")
    numbers = Array.new

    file.each_line do |line|
        aux = Array.new()
        line.each_char do |char|
            if char.to_i != 0
                aux.push(char)
            end
        end
        numbers.push([aux[0], aux[aux.length-1]].join)
    end
    file.close()
    
    sum = numbers.map(&:to_i).reduce(:+)
    puts "The sum is: #{sum}"
end